Shape_detection/examples/Shape_detection/region_growing_cylinders_on_point_set_3.cpp

```diff
@@ -0,0 +1,97 @@
+#include <fstream>
+#include <CGAL/Real_timer.h>
+#include <CGAL/Random.h>
+#include <CGAL/Simple_cartesian.h>
+
+#include <CGAL/Point_set_3.h>
+#include <CGAL/Point_set_3/IO.h>
+
+#include <CGAL/Shape_detection/Region_growing/Region_growing.h>
+#include <CGAL/Shape_detection/Region_growing/Region_growing_on_point_set.h>
+
+#include <boost/iterator/function_output_iterator.hpp>
+
+
+using Kernel = CGAL::Simple_cartesian<double>;
+using Point_3 = Kernel::Point_3;
+using Vector_3 = Kernel::Vector_3;
+
+using Point_set = CGAL::Point_set_3<Point_3>;
+using Point_map = typename Point_set::Point_map;
+using Normal_map = typename Point_set::Vector_map;
+
+namespace Shape_detection = CGAL::Shape_detection::Point_set;
+
+using Neighbor_query = Shape_detection::K_neighbor_query
+  <Kernel, Point_set, Point_map>;
+using Region_type = Shape_detection::Least_squares_cylinder_fit_region
+  <Kernel, Point_set, Point_map, Normal_map>;
+using Region_growing = CGAL::Shape_detection::Region_growing
+  <Point_set, Neighbor_query, Region_type>;
+
+int main (int argc, char** argv)
+{
+  std::ifstream ifile (argc > 1 ? argv[1] : "data/cube.pwn");
```

Comment:
Do not you need to load here something else than the cube data set? As far as I remember, it contains not only cylinders so we run into an issue of detecting multiple objects with RG, which does not sell well the algorithm.